cli/running/watchdog.go

```diff
@@ -72,23 +74,32 @@ func (wd *Watchdog) Start() {
 			break
 		}
 		wd.logger = wd.Instance.logger
-		// Start the Instance and forwarding signals (except  SIGINT and SIGTERM)
-		wd.startSignalHandling()
-		wd.stopMutex.Lock()
-		if !wd.stopped {
+
+		wd.watchdogStateMutex.Lock()
+		if !wd.shouldStop {
```

Comment:
Check for `shouldStop` and return if true. Else is not required in this case.
There is no need to have mutex locked during starting an instance. Mutex lock is required only for setting/getting the flag.

Reply:
> There is no need to have mutex locked during starting an instance. Mutex lock is required only for setting/getting the flag.

But what if I receive a signal between starting the instance and setting the flag `started` in true? I mean this should happen atomic.

Reply:
Starting application does not have to be atomic. We will set shouldStop to true in this case. And we will check this flag after application is already started.
Ok, I think current variant of locking will not have a big impact on anything. Closing this comment.